### PR TITLE
Convert pose estimation data (.csv, .pickle)

### PR DESCRIPTION
**Pull request by weiglszonja**
Add `NeurotensinDeepLabCutInterface` to convert pose estimation data in legacy DLC format (pose estimation data in .CSV file, config in .pickle file)

### CODE REVIEW

File: src/tye_lab_to_nwb/neurotensin_valence/behavior/neurotensindeeplabcutdatainterface.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            # The parameters for the pose estimation container
+            pose_estimation_kwargs = dict(
+                pose_estimation_series=pose_estimation_series,
+                # frame_dimensions are in height x width, but for NWB it should be width x height
+                dimensions=np.array([pose_estimation_config["data"]["frame_dimensions"][::-1]], dtype=np.uint16),
+                scorer=pose_estimation_config["data"]["Scorer"],
+                source_software="DeepLabCut",
+                nodes=[pose_estimation_metadata[column_name]["name"] for column_name in pose_estimation_metadata],
+                edges=np.array([(0, 1), (0, 2), (2, 3), (1, 3), (5, 6), (5, 7), (5, 8), (5, 9)], dtype=np.uint8),
+            )
```
Comment (weiglszonja) [inline src/tye_lab_to_nwb/neurotensin_valence/behavior/neurotensindeeplabcutdatainterface.py:104]: edges should move out here, probably to metadata

File: src/tye_lab_to_nwb/neurotensin_valence/behavior/neurotensindeeplabcutdatainterface.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            for column_name in pose_estimation_metadata:
+                pose_estimation_series_data = pose_estimation_data[column_name]
+
+                pose_estimation_series.append(
+                    PoseEstimationSeries(
+                        name=pose_estimation_metadata[column_name]["name"],
+                        description=pose_estimation_metadata[column_name]["description"],
+                        data=pose_estimation_series_data[["x", "y"]].values,
+                        unit="pixels",
+                        reference_frame="(0,0,0) corresponds to bottom left corner of the cage.",
```
Comment (CodyCBakerPhD) [inline src/tye_lab_to_nwb/neurotensin_valence/behavior/neurotensindeeplabcutdatainterface.py:89]:
```suggestion
                        reference_frame="(0,0) corresponds to bottom left corner of the cage.",
```

Was this confirmed with respect to the port orientation? I.e., if the port is in the bottom right, the subject (x,y) position  are both positive, traversing the positive x and y axes

**Comment by CodyCBakerPhD** [comment]
Two minor comments on metadata, the rest looks good

File: src/tye_lab_to_nwb/neurotensin_valence/behavior/neurotensindeeplabcutdatainterface.py (hunk 2 above)
Comment (weiglszonja) [reply src/tye_lab_to_nwb/neurotensin_valence/behavior/neurotensindeeplabcutdatainterface.py:89]: Plotting the port coordinates: 
port
and the camera position that should correspond to "top left":
topleft
and the "bottom right":
botright

These are the original values from the CSV, however it seems like the "y" axis should be flipped?

Original positions:
orig_visual_points

Flipped "y" axis:
visual_points
Snapshot from the labelled video:
Screenshot 2023-02-27 at 12 21 52

Comment (weiglszonja) [reply src/tye_lab_to_nwb/neurotensin_valence/behavior/neurotensindeeplabcutdatainterface.py:89]: @CodyCBakerPhD 

The positions in the CSV can be overlayed on the labeled video, the unlabelled video has a different length (num_frames=339811) compared to the labeled video (num_frames=169907):

points50


points500

points5000


points20000

`(0,0)` shows up in the top left corner when drawing it on the frame.

Comment (weiglszonja) [reply src/tye_lab_to_nwb/neurotensin_valence/behavior/neurotensindeeplabcutdatainterface.py:89]: > Was this confirmed with respect to the port orientation? I.e., if the port is in the bottom right, the subject (x,y) position are both positive, traversing the positive x and y axes

```suggestion
                        reference_frame="(0,0) corresponds to the top left corner of the cage.",
```

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**